sdk/python/feast/feature_store.py

```diff
@@ -504,43 +522,6 @@ def _apply_diffs(
             infra_diff: The diff between the current infra and the desired infra.
             new_infra: The desired infra.
         """
-        entities_to_update = [
-            fco_diff.new_fco
-            for fco_diff in registry_diff.fco_diffs
-            if fco_diff.fco_type == FeastObjectType.ENTITY
-            and fco_diff.transition_type
-            in [TransitionType.CREATE, TransitionType.UPDATE]
-        ]
-        views_to_update = [
-            fco_diff.new_fco
-            for fco_diff in registry_diff.fco_diffs
-            if fco_diff.fco_type == FeastObjectType.FEATURE_VIEW
-            and fco_diff.transition_type
-            in [TransitionType.CREATE, TransitionType.UPDATE]
-        ]
-        odfvs_to_update = [
-            fco_diff.new_fco
-            for fco_diff in registry_diff.fco_diffs
-            if fco_diff.fco_type == FeastObjectType.ON_DEMAND_FEATURE_VIEW
-            and fco_diff.transition_type
-            in [TransitionType.CREATE, TransitionType.UPDATE]
-        ]
-        request_views_to_update = [
-            fco_diff.new_fco
-            for fco_diff in registry_diff.fco_diffs
-            if fco_diff.fco_type == FeastObjectType.REQUEST_FEATURE_VIEW
-            and fco_diff.transition_type
-            in [TransitionType.CREATE, TransitionType.UPDATE]
-        ]
-
-        # TODO(felixwang9817): move validation logic into _plan.
-        # Validate all feature views and make inferences.
-        self._validate_all_feature_views(
-            views_to_update, odfvs_to_update, request_views_to_update
-        )
-        self._make_inferences(entities_to_update, views_to_update, odfvs_to_update)
-
-        # Apply infra and registry changes.
         infra_diff.update()
```

Comment:
non-blocking, but another cleanup would be to have InfraObject#update -> InfraObject#create or something since it only right now creates the object